lib/translator/inquiry.cc

```diff
@@ -0,0 +1,289 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "inquiry.h"
+#include "string.h"
+
+namespace translator {
+
+// Creates and validates a Inquiry Command struct
+StatusCode RawToScsiCommand(absl::Span<const uint8_t> raw_cmd,
+                            scsi_defs::InquiryCommand &cmd) {
+  if (raw_cmd.empty()) {
+    DebugLog("buffer is empty or nullptr\n");
+    return StatusCode::kInvalidInput;
+  }
+
+  uint8_t raw_opcode = raw_cmd[0];
+  if (raw_opcode > 0xaf) {
+    DebugLog("raw opcode is out of range: %x", raw_opcode);
+    return StatusCode::kInvalidInput;
+  }
+
+  scsi_defs::OpCode opcode = static_cast<scsi_defs::OpCode>(raw_opcode);
+  if (!strcmp(ScsiOpcodeToString(opcode), "INVALID OPCODE")) {
```

Comment:
Is there any way to get the same level of validation without having to perform a string comparison?

Reply:
ah yep, this is unnecessary. the check `  if (opcode != scsi_defs::OpCode::kInquiry)` suffices.